design/application-abi.md

```diff
@@ -12,20 +12,29 @@ include.
 Current Unstable ABI
 --------------------
 
-The current WASI unstable ABI specifies only two exports from a WASI
-application:
+There are two kinds of modules:
 
-- `_start` - the program entry point
-- `memory` - linear memory used by the program
+ - A *command* exports a function named `_start`, with no arguments and no return
```

Comment:
Partially off topic, but why not pass some configuration data to `_start`?

For example, the function could receive the command line args, the allowed paths, etc. These would be through interface types, of course.

Reply:
Yes, exactly. There are a lot of things we could do here, but we need interface types to do them well. The current `argv`/`environ` interfaces we have now aren't super amazing, and `_start` isn't a great name, and so on, but at the moment it all works well enough that it doesn't seem worth breaking compatibility until we have a big reason for it, like interface types.